src/guardian/duplicates/keys-duplication-checker.service.ts

```diff
@@ -241,13 +242,20 @@ export class KeysDuplicationCheckerService {
     key: string,
     blockData: BlockData,
   ): Promise<SigningKeyEvent[]> {
-    const { events } =
-      await this.signingKeyEventsCacheService.getUpdatedSigningKeyEvents(
+    const eventsGroup =
+      await this.signingKeysRegistryService.getUpdatedSigningKeyEvents(
         key,
         blockData.blockNumber,
         blockData.blockHash,
       );
-    return events;
+
+    if (!eventsGroup.isValid) {
```

Comment:
it will break whole keys check
instead based on duplicates logic if we are not sure that events cache correct we should put this events in a list of unresolvedDuplicates

Reply:
fixed